include/rmm/mr/default_memory_resource.hpp

```diff
@@ -51,5 +51,43 @@ device_memory_resource* get_default_resource();
 device_memory_resource* set_default_resource(
     device_memory_resource* new_resource);
 
+
+/**---------------------------------------------------------------------------*
+ * @brief gets a cuda_memory_resource
+ *
+ * A static function which will return a singleton cuda_memory_resource
+ *
+ *
+ *
+ * @return device_memory_resource* a pointer to the static
+ * cuda_memory_resource
+ *---------------------------------------------------------------------------**/
+device_memory_resource* initial_resource();
```

Comment:
For completeness, lets add a `cuda_resource()` as well, and then have `initial_resource` just return `cuda_resource()`.